pkg/ssh/ssh_keys.go

```diff
@@ -0,0 +1,110 @@
+package ssh
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/run"
+	"github.com/cli/safeexec"
+)
+
+type Context struct {
+	ConfigDir string
+	KeygenExe string
+}
+
+type KeyPair struct {
+	PublicKeyPath  string
+	PrivateKeyPath string
+}
+
+type KeyAlreadyExistsError struct {
+	keyPath string
+}
+
+func (err *KeyAlreadyExistsError) Error() string { return "SSH key already exists: " + err.keyPath }
+
+func (c *Context) LocalPublicKeys() ([]string, error) {
+	sshDir, err := c.sshDir()
+	if err != nil {
+		return nil, err
+	}
+
+	return filepath.Glob(filepath.Join(sshDir, "*.pub"))
+}
+
+func (c *Context) HasKeygen() bool {
+	_, err := c.findKeygen()
+	return err == nil
+}
+
+func (c *Context) GenerateSSHKey(keyName string, passphrase string) (*KeyPair, error) {
+	keygenExe, err := c.findKeygen()
+	if err != nil {
+		return nil, fmt.Errorf("could not find keygen executable")
```

Comment:
Do we not want to wrap the underlying error from `findKeygen` ?

Reply:
I would say just return the original error without wrapping. It will be descriptive enough

Reply:
Just used the original error